Source/WebCore/dom/Observable.cpp

```diff
@@ -71,28 +91,12 @@ void Observable::subscribe(ScriptExecutionContext& context, std::optional<Observ
     }
 }
 
-Ref<Subscriber> Observable::makeSubscriber(ScriptExecutionContext& context, std::optional<ObserverUnion> observer)
-{
-    if (observer.has_value()) {
-        return WTF::switchOn(
-            observer.value(),
-            [&](RefPtr<JSSubscriptionObserverCallback>& next) {
-                return Subscriber::create(context, next);
-            },
-            [&](SubscriptionObserver& subscription) {
-                return Subscriber::create(context, subscription.next, subscription.error, subscription.complete);
-            }
-        );
-    }
-
-    return Subscriber::create(context);
-}
-
 WTF_MAKE_ISO_ALLOCATED_IMPL(Observable);
 
-Observable::Observable(Ref<SubscriberCallback> callback)
-    : m_subscriber(callback)
+Observable::Observable(RefPtr<SubscriberCallback> subscriber)
+    : m_subscriber(subscriber)
 {
+    ASSERT(m_subscriber);
```

Comment:
Can an observable be without a subscriber callback? If not, this should be a `Ref<Subscriber>`

Reply:
For now it cannot, follow up PRs which add new operators will pass in a lambda instead of the `Subscriber` object.

Reply:
Still, if the invariant is that this cannot be null then instead of a RefPtr it should be a Ref.